fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/FieldValue.java

```diff
@@ -169,4 +170,9 @@ private static Type resolveTypeForPath(@Nonnull final Type inputType, @Nonnull f
         }
         return currentType;
     }
+
+    @Nonnull
+    public KeyExpression toKeyExpression() {
```

Comment:
This is dangerous as the `FieldValue` as it only captures the path contained in here, i.e. the source the field value operates on may not be the root message. I would suggest to either allow a prefix to be passed in (which then can be empty) or to not put his code here.